impeller/entity/contents/vertices_contents.cc

```diff
@@ -40,57 +51,75 @@ void VerticesContents::SetBlendMode(BlendMode blend_mode) {
 bool VerticesContents::Render(const ContentContext& renderer,
                               const Entity& entity,
                               RenderPass& pass) const {
-  auto& host_buffer = pass.GetTransientsBuffer();
-  auto vertex_type = geometry_->GetVertexType();
+  if (blend_mode_ == BlendMode::kClear) {
+    return true;
+  }
+  auto dst_contents = std::make_shared<VerticesColorContents>(*this);
+
+  if (blend_mode_ < BlendMode::kScreen) {
+    auto contents = ColorFilterContents::MakeBlend(
+        blend_mode_,
+        {FilterInput::Make(dst_contents), FilterInput::Make(src_contents_)});
```

Comment:
Hoo boy 😅, so the edge blending issue is caused by both the source and destination inputs to the blend being cut out with the same geometry and antialiased individually via MSAA resolves, and so the edge pixels of both the source/destination have varying translucency, affecting the blend behavior around the edges.

Unless we want to get in the business of storing MSAA textures (_enormous_ GPU performance hit, but we very well may want to do this at some point), fixing this in a "correct" way such that the result is both correctly blended and properly antialiased is very much non-trivial, and I think the best option will be to compromise fidelity a bit. 😓

TBH, I would just go for disable antialiasing in this niche "the backdrop colors need to be blended before we blend in the source" situation...
This can be done by adding an `msaa_enabled` field (defaults to true) to `ContentsFilterInput` and `Contents::RenderToSnapshot`. If it's false, then `RenderToSnapshot` will be expected use a non-MSAA pass to render the contents to a texture.

I've actually had something like this on my todo for experimental optimizations for a while (for cases where the user wouldn't notice that things aren't antialiased) -- we'll be able to reuse this feature for other things later on. Eventually, we'll likely want to support expressing whether MSAA should be used or not in the `Snapshot` as well, but we can worry about that kind of thing later on if/when we need it, and simply ignore the new parameter for the `TextureContents` snapshot deferral cases that exist today.

Reply:
That sounds like a reasonable approach. FWIW I don't belive drawVertices is anti aliased at all in the Skia backend, and as long as retain the anti aliasing outside of the weird blend cases I think we'll be fine here